airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java

```diff
@@ -51,6 +54,14 @@ public class PostgresQueryUtils {
             txid_snapshot_xmin(txid_current_snapshot()) AS xmin_raw_value;
       """;
 
+  public static final String CTID_FULL_VACUUM_IN_PROGRESS_QUERY =
+      """
+      SELECT phase FROM pg_stat_progress_cluster WHERE command = 'VACUUM FULL' AND relid=to_regclass('%s')::oid    
```

Comment:
Seems like pg_stat_progress_cluster was added in Postgres 12? https://pgpedia.info/p/pg_stat_progress_cluster. What's the behavior of this query for versions prior to that but are supported by Airbyte (PG 9.4 - 12)?

Reply:
Great catch! Thank you
What I think will work best in the real sync is if we can gracefully fail if `pg_stat_progress_cluster` function wasn't supported and fallback to standard non-ctid sync.
For the purpose of this logging only call I'll make sure we're showing a warning which doesn't affect the overall sync.
I'm testing with an older pg to understand what JDBC is giving us in that case.

Reply:
Thanks! Just wanted to make sure it didn't cause a NPE or something

Reply:
Very good point. and another thing to look for when reading using ctid

Reply:
Another quick thing you can do is to change the docker image we're using here to test the `logPreSyncDebugData` functionality pre-12 : Not blocking, but thought I'd mention :  https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/PostgresSourceTest.java#LL143C46-L143C46